core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -199,8 +199,7 @@ private[spark] object ReliableCheckpointRDD extends Logging {
 
     val finalOutputName = ReliableCheckpointRDD.checkpointFileName(ctx.partitionId())
     val finalOutputPath = new Path(outputDir, finalOutputName)
-    val tempOutputPath = new Path(outputDir,
-      s".$finalOutputName-attempt-${ctx.stageAttemptNumber()}-${ctx.attemptNumber()}")
+    val tempOutputPath = new Path(outputDir, s".$finalOutputName-attempt-${ctx.taskAttemptId()}")
```

Comment:
Ya. It looks simpler.

Reply:
BTW, do we need `-attempt-`?

Reply:
```
private[spark] class TaskDescription(
    val taskId: Long,
    val attemptNumber: Int,
...
```
I believe `taskAttempId` is some kind of historic name in `TaskContext`.